### PR TITLE
Fix IPC command length limit

### PR DESCRIPTION
**Pull request by liuzhe-lz**
The protocol for non-trialkeeper services is changed so that when a command exceeds 999999 length, it will be splitted into multiple parts.
For example, an "initialize" (`IN`) command with 3000000 letter `x` will be splitted to:
  - `INEXCEED` + 999999 letter `x`
  - `_CEXCEED` + 999999 letter `x` (`_C` stands for "continue")
  - `_CEXCEED` + 999999 letter `x`
  - `_E000003xxx` (`_E` stands for "end")

### CODE REVIEW

File: src/nni_manager/core/ipcInterface.ts [hunk 1]
```diff
@@ -22,13 +22,22 @@ const ipcIncomingFd: number = 4;
  * @returns binary command data
  */
 function encodeCommand(commandType: string, content: string): Buffer {
-    const contentBuffer: Buffer = Buffer.from(content);
-    if (contentBuffer.length >= 1_000_000) {
-        throw new RangeError('Command too long');
+    let contentBuffer: Buffer = Buffer.from(content);
```
Comment (QuanluZhang) [inline src/nni_manager/core/ipcInterface.ts:25]: suggest to add docstring to explain the protocol

File: src/sdk/pynni/nni/protocol.py [hunk 2]
```diff
@@ -43,9 +47,16 @@ def send(command, data):
...
+        if len(data) <= _max_msg_len:
+            msg = b'%b%06d%b' % (command.value, len(data), data)
+        else:
+            msg = b'%b______%b' % (command.value, data[:_max_msg_len])
+            data = data[_max_msg_len:]
+            while len(data) > _max_msg_len:
+                msg += b'_C999999%b' % data[:_max_msg_len]
+                data = data[_max_msg_len:]
+            msg += b'_E%06d%b' % (len(data), data)
         _out_file.write(msg)
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/protocol.py:60]: if we can write data with any large content length, why not simply increase the number of bits for representing data length?

File: src/nni_manager/.eslintrc [hunk 3]
```diff
@@ -23,7 +23,8 @@
         "@typescript-eslint/consistent-type-assertions": 0,
         "@typescript-eslint/no-inferrable-types": 0,
         "no-inner-declarations": 0,
-        "@typescript-eslint/no-var-requires": 0
+        "@typescript-eslint/no-var-requires": 0,
+        "no-constant-condition": ["error", {"checkLoops": false}]
```
Comment (liuzhe-lz) [inline src/nni_manager/.eslintrc:27]: This rule is to allow `while (true)`

File: src/nni_manager/core/ipcInterface.ts (hunk 1 above)
Comment (liuzhe-lz) [reply src/nni_manager/core/ipcInterface.ts:29]: The doc will be added to wiki.

**Pull request closed without merging**